File: pythonCode/dilation.py
import cv2
import sys

#read the image to be dilated
image = cv2.imread("../assets/dilation.png")

#check if the input image exits
if image is None:
    print("Input image not found")
    sys.exit()

#Create a structiong element/kernel which will be used for dilation
dilationSize = 6
element = cv2.getStructuringElement(cv2.MORPH_CROSS, (2*dilationSize + 1, 2* dilationSize + 1), (dilationSize, dilationSize))

#Apply dilate function on input image. Dilation will increase brightness, First Parameter is the original image, 
#second is the dilated image
dilatedImage = cv2.dilate(image, element)

#Create windows to diaplay images
cv2.namedWindow("input image", cv2.WINDOW_NORMAL)
cv2.namedWindow("dilated image", cv2.WINDOW_NORMAL)

#Display the images
cv2.imshow("input image", image)
cv2.imshow("dilated image", dilatedImage)

#Press esc on keyboard to exit the program
cv2.waitKey(0)

#close all the opened windows
cv2.destroyAllWindows()

File: cPlusPlusCode/dilation.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace std;
using namespace cv;

int main(){
    //Read image to dilate
    Mat image = imread("../assets/dilation.png");
    
    //Create a structuring element/kernel
    int dilationSize = 6;
    Mat element = getStructuringElement(MORPH_CROSS, Size(2*dilationSize + 1, 2*dilationSize + 1), Point(dilationSize, dilationSize));

    Mat dilatedImage;

    //Apply dilate function on input image. Dilation will increase brightness, First Parameter is the original image, 
    //second is the dilated image
    dilate(image, dilatedImage, element);

    //create windows to show image
    namedWindow("input image", WINDOW_NORMAL);
    namedWindow("dilated image", WINDOW_NORMAL);

    //Display input and dilated image
    imshow("input image", image);
    imshow("dilated image", dilatedImage);

    //Press esc on keyboard to exit the program
    waitKey(0);

    //Close all the opened windows
    destroyAllWindows();

    return 0;
}